### PR TITLE
Adding Unit Tests for Request Processing engine

### PR DESCRIPTION
**Pull request by kpratyus**
The Unit tests are added for the issue https://github.com/open-telemetry/opentelemetry-cpp-contrib/issues/133

### CODE REVIEW

File: instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp [hunk 1]
```diff
@@ -48,10 +48,66 @@ MATCHER_P(HasStringVal, value, "") {
...
+
+	using namespace opentelemetry;
+	appd::core::sdkwrapper::OtelKeyValueMap argkeyValueMap = arg;
+	bool valueMatches = true;
+	for(auto argKey:argkeyValueMap){
+		
+		auto argkeyValue = argKey.second;
+		auto keyValue = argkeyValue;
+
+		if(value.find(argKey.first) != value.end()){
```
Comment (DebajitDas) [inline instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp:61]: auto itr = value.find(argKey.first);
if (itr != value.end) {
    keyValue = itr->second;
}

File: instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp [hunk 2]
```diff
@@ -48,10 +48,66 @@ MATCHER_P(HasStringVal, value, "") {
   return opentelemetry::nostd::get<opentelemetry::nostd::string_view>(arg) == value;
 }
 
+MATCHER_P(HasMapVal, value, "") {
+
+	using namespace opentelemetry;
+	appd::core::sdkwrapper::OtelKeyValueMap argkeyValueMap = arg;
+	bool valueMatches = true;
+	for(auto argKey:argkeyValueMap){
```
Comment (DebajitDas) [inline instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp:56]: auto &

File: instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp [hunk 3]
```diff
@@ -197,9 +253,10 @@ TEST(TestRequestProcessingEngine, StartInteraction)
 	std::unordered_map<std::string, std::string> emptyHeaders;
 
 	// sdkwrapper's create span function should be called
+	using testing::_;
 	EXPECT_CALL(*sdkWrapper, CreateSpan("module_phase",
 		appd::core::sdkwrapper::SpanKind::CLIENT,
-		keyValueMap, emptyHeaders)).
+		_, emptyHeaders)).
```
Comment (DebajitDas) [inline instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp:259]: use matcher

Comment (kpratyus) [reply instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp]: Resolved

File: instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp (hunk 2 above)
Comment (kpratyus) [reply instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp]: Added

File: instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp (hunk 1 above)
Comment (kpratyus) [reply instrumentation/otel-webserver-module/test/unit/RequestProcessingEngine_test.cpp]: Changed

**Review by DebajitDas — APPROVED** [review]

**Pull request merged**